File: frontend/src/types/maintenance.ts
/**
 * Maintenance history types.
 * PROTOTYPE DEMO — fictional data, not real military / DELIIS records.
 */

export interface AircraftInfo {
  aircraft_id: string;
  aircraft_type: string;
  aircraft_number: string;
  total_flight_hours: number;
  display_name?: string | null;
  alias_of?: string | null;
  maintenance_count?: number;
}

export interface MaintenanceHistoryRecord {
  maintenance_id: string;
  aircraft_id: string;
  maintenance_date: string;
  flight_hours: number;
  fault_code: string;
  system_category: string;
  component: string;
  symptom: string;
  detected_value?: string | null;
  normal_range?: string | null;
  severity: string;
  diagnosis: string;
  root_cause: string;
  maintenance_action: string;
  replaced_part?: string | null;
  technician_note?: string | null;
  maintenance_result: string;
  recurrence: boolean;
  reference_manual?: string | null;
  symptom_code?: string | null;
  system_code?: string | null;
  created_at: string;
  is_dummy?: boolean;
}

export interface SimilarMaintenanceItem {
  record: MaintenanceHistoryRecord;
  similarity: number;
  similarity_percent: number;
}

export interface MaintenanceHistoryCreate {
  aircraft_id: string;
  maintenance_date?: string;
  flight_hours?: number;
  fault_code?: string;
  system_category?: string;
  component?: string;
  symptom: string;
  detected_value?: string;
  normal_range?: string;
  severity?: string;
  diagnosis?: string;
  root_cause: string;
  maintenance_action: string;
  replaced_part?: string;
  technician_note?: string;
  maintenance_result: string;
  recurrence?: boolean;
  reference_manual?: string;
  symptom_code?: string;
  system_code?: string;
}

export interface MaintenanceHistoryStats {
  total: number;
  last_30_days: number;
  by_system: Record<string, number>;
  top_faults: { symptom: string; count: number }[];
  recurrence_count: number;
  recent: MaintenanceHistoryRecord[];
}

export interface MaintenanceHistoryFilters {
  aircraft_id?: string;
  system_category?: string;
  component?: string;
  fault_code?: string;
  symptom?: string;
  maintenance_result?: string;
  severity?: string;
  date_from?: string;
  date_to?: string;
}

File: backend/app/schemas/maintenance.py
"""Maintenance history schemas. Prototype / fictional data only — not DELIIS."""

from __future__ import annotations

from pydantic import BaseModel, Field


class AircraftInfo(BaseModel):
    aircraft_id: str
    aircraft_type: str
    aircraft_number: str
    total_flight_hours: float
    display_name: str | None = None
    alias_of: str | None = None


class MaintenanceHistoryRecord(BaseModel):
    maintenance_id: str
    aircraft_id: str
    maintenance_date: str
    flight_hours: float = 0.0
    fault_code: str = ""
    system_category: str = ""
    component: str = ""
    symptom: str
    detected_value: str | None = None
    normal_range: str | None = None
    severity: str = "Warning"
    diagnosis: str = ""
    root_cause: str = ""
    maintenance_action: str = ""
    replaced_part: str | None = None
    technician_note: str | None = None
    maintenance_result: str = ""
    recurrence: bool = False
    reference_manual: str | None = None
    symptom_code: str | None = None
    system_code: str | None = None
    created_at: str
    is_dummy: bool = True


class SimilarMaintenanceItem(BaseModel):
    record: MaintenanceHistoryRecord
    similarity: float = Field(ge=0.0, le=1.0)
    similarity_percent: int = Field(ge=0, le=100)


class MaintenanceHistoryCreate(BaseModel):
    aircraft_id: str
    maintenance_date: str | None = None
    flight_hours: float | None = None
    fault_code: str | None = None
    system_category: str | None = None
    component: str | None = None
    symptom: str
    detected_value: str | None = None
    normal_range: str | None = None
    severity: str = "Warning"
    diagnosis: str | None = None
    root_cause: str
    maintenance_action: str
    replaced_part: str | None = None
    technician_note: str | None = None
    maintenance_result: str
    recurrence: bool = False
    reference_manual: str | None = None
    symptom_code: str | None = None
    system_code: str | None = None


class MaintenanceHistoryStats(BaseModel):
    total: int
    last_30_days: int
    by_system: dict[str, int]
    top_faults: list[dict[str, str | int]]
    recurrence_count: int
    recent: list[MaintenanceHistoryRecord]
